core/ast_node.js

```diff
@@ -177,12 +174,13 @@ Blockly.ASTNode.createWorkspaceNode = function(workspace, wsCoordinate) {
 
 /**
  * Parse the optional parameters.
- * @param {?Object} params The user specified parameters.
+ * @param {Object=} params The user specified parameters.
```

Comment:
This isn't optional: you're returning early if you don't get params.  It *is* nullable.

Reply:
fixed